src/PiWeb.Import.Sdk/Modules/ImportFormat/AttributeMappingConfiguration.cs

```diff
@@ -0,0 +1,37 @@
+﻿#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+using System.Collections.Generic;
+
+namespace Zeiss.PiWeb.Import.Sdk.Modules.ImportFormat;
+
+/// <summary>
+/// Represents the attribute mapping rules to be used for an import format.
+/// </summary>
+public class AttributeMappingConfiguration
+{
+    /// <summary>
+    /// An <see cref="AttributeMappingConfiguration"/> instance without any attribute mapping rules.
+    /// </summary>
+    public static readonly AttributeMappingConfiguration Empty = new AttributeMappingConfiguration();
+
+    /// <summary>
+    /// Specifies whether auto mapping should be applied.
+    /// The auto mapping functionality maps entity variables of the form Kxxxx automatically
+    /// to attributes with the same attribute key without explicit mapping rule.
+    /// Auto mapping for an attribute is only applied if no explicit mapping rule for this attribute exists.
+    /// </summary>
+    public bool AutoMapping { get; init; } = true;
+
+    /// <summary>
+    /// The list of mapping rules. 
+    /// </summary>
+    public IReadOnlyCollection<MappingRule> MappingRules { get; init; } = [];

```

Comment:
The read only collection does not prevent the plugin writer from changing the collection, it only prevents us from changing it in the Auto Importer. Since this should not be a performance issue, we probably should just use an immutable list here. Or if we make a defensive copy somewhere, then IEnumerable should be fine here.

Reply:
Yeah, it should be an immutable list, I'll change that.

Reply:
Done